src/main/java/com/box/sdk/BoxAPIResponse.java

```diff
@@ -48,6 +49,18 @@
      */
     public BoxAPIResponse() {
         this.connection = null;
+        this.headers = null;
+    }
+
+    /**
+     * Constructs a BoxAPIResponse with a http response code and response headers.
+     * @param responseCode http response code
+     * @param headers map of headers
+     */
+    public BoxAPIResponse(int responseCode, Map<String, String> headers) {
```

Comment:
Reusing BoxAPIResponse. It's an existing class. No need for special BoxBatchAPIResponse as the response will be same but a list instead of one response